workspace/exercise_test.go

```diff
@@ -42,3 +67,92 @@ func TestNewFromDir(t *testing.T) {
 	assert.Equal(t, "the-track", exercise.Track)
 	assert.Equal(t, "the-exercise", exercise.Slug)
 }
+
+func TestMigrateLegacyMetadataFileWithoutLegacy(t *testing.T) {
+	ws, err := ioutil.TempDir("", "fake-workspace")
+	defer os.RemoveAll(ws)
+	assert.NoError(t, err)
+
+	exercise := Exercise{Root: ws, Track: "bogus-track", Slug: "no-legacy"}
+	metadataFilepath := exercise.MetadataFilepath()
+	err = os.MkdirAll(filepath.Dir(metadataFilepath), os.FileMode(0755))
+	assert.NoError(t, err)
+
+	err = ioutil.WriteFile(metadataFilepath, []byte{}, os.FileMode(0600))
+	assert.NoError(t, err)
+
+	ok, _ := exercise.HasLegacyMetadata()
+	assert.False(t, ok)
+	ok, _ = exercise.HasMetadata()
+	assert.True(t, ok)
+
+	status, err := exercise.MigrateLegacyMetadataFile()
+	assert.Equal(t, MigrationStatus(MigrationStatusNoop), status)
```

Comment:
You shouldn't need this type cast, since MigrateLegacyMetadataFile() returns a MigrationStatus not an int.